---
title: "NC Timecourse RNA-Seq: DESeq2"
output: html_notebook
---
# Step 1: Quality Control
```{bash}
./process_RNASeq.sh
```


The First RMD notebook for the NC Timecourse RNA-Seq analysis. This document is the first of the series.
In this document, we will start from the output of 'process_RNASeq.sh', with BAMs aligned to the ENSEMBL
galGal6 genome.



Check out the MultiQC report that was generated to identify any outliers in sequence or alignment quality.


### Load the libraries required
This analysis was performed on a 16 core machine running Ubuntu 18.04 LTS. Certain parameters may need to be updated based on your operating system and available CPUs.
```{r Loading Libraries, message=FALSE, include=TRUE}
library(tidyr)
library(edgeR)
library(reshape2)
library(ggplot2)
library(scales)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(dplyr)
library(Rsubread)
library(DESeq2)
library(dplyr)
library(readr)
library(ggsci)
library(purrr)
library(BiocParallel)
library(org.Gg.eg.db)
library(ashr)
library(BiocGenerics)
library(stringr)
register(SnowParam(16)) # Set this to number of CPUs.
```

### Read in the results of the featureCounts
```{r Reading in featureCounts results}
counts_summary <- read.delim("./Imports/featureCounts.txt.summary", row.names = 1)
counts_summary <- rbind(counts_summary, percent_aligned = (counts_summary[1,]/colSums(counts_summary))*100)

featureCounts <- read.delim("~/local_git/NC_Timecourse/RNA-Seq/Imports/featureCounts.txt", comment.char="#")

#Remove the gene information to just get the counts matrix.
dat <- featureCounts[,7:56]
rownames(dat) <- featureCounts$Geneid

#Clean sample names
sample_names <- colnames(featureCounts[7:56])
sample_names <- str_sub(sample_names, 7,-5)
sample_names <- word(sample_names, sep = "_", 5, -3)
sample_names <- gsub("\\.", "_", sample_names)
sample_names <- gsub("MR[0-9]+_", "", sample_names)
sample_names <- gsub("MR[0-9]+R_", "", sample_names)
colnames(dat) <- sample_names

# Generating metadata file
colData <- data.frame(row.names=colnames(dat), data=colnames(dat))
colData <- colData %>% mutate(Condition = factor(ifelse(grepl(pattern = "pos|pGFP", x = data), "NC","WE")))
colData <- colData %>% mutate(Time = factor(x = str_extract(word(colData$data, sep = "_",1,1), pattern ="[0-9]+"), levels = c(6,8,10,12,14,16)))
colData <- colData %>% mutate(Group = factor(paste0(Condition,"_",Time)))

head(colData)
```

### Begin DESeq2 Analysis
Note, this is pairwise comparison between NC and WE cells at each timepoint, additional analysis will be done for detecting genes varying through time. [ImpulseDE2](https://bioconductor.org/packages/release/bioc/vignettes/ImpulseDE2/inst/doc/ImpulseDE2_Tutorial.html)
```{r DESeq2}
dds <- DESeqDataSetFromMatrix(countData = dat,
                              colData = colData,
                              design= ~0 + Group)

dds <- DESeq(dds, parallel = TRUE)
results(dds, parallel = TRUE)

#Get the contrasts from the design
resultsNames(dds)

#Check out the counts
bp1 <-boxplot(log10(counts(dds,normalized=TRUE)+1))
bp1

#What does the dispersion look like?
disp_est <- plotDispEsts(dds)

```

