---
title: "NC Timecourse RNA-Seq: DESeq2"
output: html_notebook
---
# Step 1: Quality Control
The First RMD notebook for the NC Timecourse RNA-Seq analysis. This document is the first of the series.
In this document, we will start from the output of 'process_RNASeq.sh', with BAMs aligned to the ENSEMBL
galGal6 genome.



Check out the MultiQC report that was generated to identify any outliers in sequence or alignment quality.


### Load the libraries required
This analysis was performed on a 16 core machine running Ubuntu 18.04 LTS. Certain parameters may need to be updated based on your operating system and available CPUs.
```{r Loading Libraries, message=FALSE, include=TRUE}
library(tidyr)
library(edgeR)
library(reshape2)
library(ggplot2)
library(scales)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(dplyr)
library(Rsubread)
library(DESeq2)
library(dplyr)
library(readr)
library(ggsci)
library(purrr)
library(BiocParallel)
library(org.Gg.eg.db)
library(ashr)
library(BiocGenerics)
register(SnowParam(16)) # Set this to number of CPUs.
```

### Read in the results of the featureCounts
```{r Reading in featureCounts results}
rownames(FeatureCounts) <- FeatureCounts[,1]
dat <- FeatureCounts[7:52]

rownames(FeatureCounts_HH12v4) <- FeatureCounts_HH12v4[,1]
dat2 <- FeatureCounts_HH12v4[7:10]

dat_merged <- cbind(dat,dat2)

Sample.Names <- read.table("Sample Names_total.txt", quote="\"", comment.char="")
colnames(dat_merged) <- Sample.Names$V1
```

### Begin DESeq2 Analysis
```{r DESeq2}

```

